src/engine/SpatialJoin.cpp

```diff
@@ -386,15 +390,18 @@ SpatialJoin::prepareJoin() const {
 
   // Size of output table
   size_t numColumns = getResultWidth();
-  return std::make_tuple(resLeft, resRight, leftJoinCol, rightJoinCol,
-                         numColumns);
+  return PreparedJoinParams{resLeft,        keepAliveLeft, resRight,
+                            keepAliveRight, leftJoinCol,   rightJoinCol,
```

Comment:
You can std::move(...) the keepalive objects (the performance gain for shared_ptr is probably not important here, but for your mindset as a programmer.